dotCMS/src/main/java/com/dotcms/cms/login/LoginServiceAPIImpl.java

```diff
@@ -0,0 +1,537 @@
+                    final User user, final boolean rememberMe) {
+
+        int jwtMaxAge = rememberMe ? Config.getIntProperty(JSON_WEB_TOKEN_DAYS_MAX_AGE,
+                        JSON_WEB_TOKEN_DAYS_MAX_AGE_DEFAULT) : -1;
+
+        this.doRememberMe(req, res, user, jwtMaxAge);
+    } // doRememberMe.
+
+    @Override
+    public void doRememberMe(final HttpServletRequest req, final HttpServletResponse res,
+                    final User user, final int maxAge) {
+
+        try {
+
+            this.processJsonWebToken(req, res, user, maxAge);
+        } catch (Exception e) {
+
+            Logger.debug(this, e.getMessage(), e);
+        }
+    } // doRememberMe.
+
+
+    private boolean _doCookieLogin(String encryptedId, HttpServletRequest request,
+                    HttpServletResponse response) {
+
+        try {
+            String decryptedId = PublicEncryptionFactory.decryptString(encryptedId);
+            /* Custom Code */
+            User user = null;
+            if (Validator.isEmailAddress(decryptedId))
+                user = userAPI.loadByUserByEmail(decryptedId, systemUser, false);
+            else
+                user = userAPI.loadUserById(decryptedId, systemUser, false);
+            /* End of Custom Code */
+            try {
+                String userName = (authByEmail) ? user.getEmailAddress() : user.getUserId();
+
+                return _doLogin(userName, null, true, request, response, true);
+            } catch (Exception e) { // $codepro.audit.disable logExceptions
+                SecurityLogger.logInfo(this.getClass(),
+                                "An invalid attempt to login (No user found) from IP: "
+                                                + request.getRemoteAddr() + " :  " + e);
+
+                return false;
+            }
+        } catch (Exception e) {
+            SecurityLogger.logInfo(this.getClass(), "Auto login failed (No user found) from IP: "
+                            + request.getRemoteAddr() + " :  " + e);
+
+            doLogout(request, response);
+
+            return false;
+
+        }
+    }
+
+
+
+    @Override
+    public boolean doCookieLogin(final String encryptedId, final HttpServletRequest request,
+                    final HttpServletResponse response) {
+        // note: keep in mind we are doing BE and FE login, not sure if this is right
+        final boolean doCookieLogin = _doCookieLogin(encryptedId, request, response);
+
+
+
+        return doCookieLogin;
+    }
+
+    @Override
+    public boolean doLogin(final String userName, final String password, final boolean rememberMe,
+                    final HttpServletRequest request, final HttpServletResponse response)
+                    throws NoSuchUserException {
+
+        return _doLogin(userName, password, rememberMe, request, response, false);
+    }
+
+
+
+    private boolean _doLogin(final String userIdOrEmail, final String password,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)